Sources/MongoSwift/MongoError.swift

```diff
@@ -251,6 +251,18 @@ private func parseMongocError(_ error: bson_error_t, reply: Document?) -> MongoE
             message: message,
             errorLabels: errorLabels
         )
+    case (MONGOC_ERROR_WRITE_CONCERN, _):
+        var writeConcernErrorLabels =
+            reply?["writeConcernError"]?.documentValue?["errorLabels"]?.arrayValue?.asArrayOf(String.self)
+        if let errorLabels = errorLabels {
+            writeConcernErrorLabels = Array(Set((writeConcernErrorLabels ?? []) + errorLabels))
+        }
+        return CommandError(
```

Comment:
The reason I introduced the new error here is because committing a transaction can result in a write concern error but no write error. Here's the error message thrown by the test **write concern error on commit** in **commit.json**:
```
{
    "writeConcernError" : {
        "code" : 100,
        "codeName" : "UnsatisfiableWriteConcern",
        "errmsg" : "Not enough data-bearing nodes",
        "errInfo" : {
            "writeConcern" : {
                "w" : 10,
                "wtimeout" : 0,
                "provenance" :
                "clientSupplied"
            }
        }
    },
    "ok" : 1.0,
    "$clusterTime" : {
        "clusterTime" : {
            "$timestamp" : {
                "t" : 1586298210,
                "i" : 17
            }
        },
        "signature" : {
            "hash" : {
                "$binary" : {
                    "base64": "AAAAAAAAAAAAAAAAAAAAAAAAAAA=",
                    "subType" : "00"
                }
            },
            "keyId" : 0
        }
    },
    "operationTime" : {
        "$timestamp" : {
            "t" : 1586298210,
            "i" : 17
        }
    }
}
```
In its current form, `extractMongoError` in `MongoError.swift` parses an error into a `WriteError` if and only if `serverReply` contains the fields `writeErrors` or `writeConcernErrors`. It doesn't look for `writeConcernError`. I've changed the logic to reflect the new field name.

Reply:
This will cause the proper `WriteError` to be constructed.

Reply:
Ah thats a really good catch. I think the `writeError_s_` comes from us funneling everything through the bulk API, and libmongoc appends an "s" to the end of the writeConcernError field because there could be many.